src/vitamin.cpp

```diff
@@ -51,7 +53,7 @@ void vitamin::load_vitamin( JsonObject &jo )
 
     vit.id_ = vitamin_id( jo.get_string( "id" ) );
     jo.read( "name", vit.name_ );
-    vit.deficiency_ = efftype_id( jo.get_string( "deficiency" ) );
+    vit.deficiency_ = efftype_id( jo.get_string( "deficiency", "null" ) );
```

Comment:
```suggestion
    vit.deficiency_ = efftype_id::NULL_ID();
    jo.read( "deficiency", vit.deficiency_ );
```

Reply:
Why is this better? Should I change the line below it too, then?

Reply:
It avoids the magic string `"null"` and uses `string_id`'s own json deserialization function, which is better for code maintenance (imagine if someone wants to change the way `efftype_id` is read. If it's read using its own deserialization function, only the deserialization function needs to be updated. If it's read by reading a string and using the string to construct `efftype_id`, every piece of code that reads an `efftype_id` has to be changed)

And yes, please also change the line below.

Reply:
Makes sense, thanks for the detailed explanation! I will try to avoid using `"null"`, I can definitely see why it would cause code maintenance issues.